src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,38 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) obj;
+            return width.equals(((Rectangle) obj).width)
```

Comment:
do we need to cast every field of obj, if we have already rectangle object, can we use it here?